src/elm/type.coffee

```diff
@@ -56,23 +57,36 @@ module.exports.ToDecimal = class ToDecimal extends Expression
 
   exec: (ctx) ->
     arg = @execArgs(ctx)
-    if arg? and typeof arg != 'undefined' then parseFloat(arg.toString()) else null
+    if arg? and typeof arg != 'undefined'
+      decimal = parseFloat(arg.toString())
+      return decimal if isValidDecimal(decimal)
+     else
+      return null
 
 module.exports.ToInteger = class ToInteger extends Expression
   constructor: (json) ->
     super
 
   exec: (ctx) ->
     arg = @execArgs(ctx)
-    if arg? and typeof arg != 'undefined' then parseInt(arg.toString()) else null
+    if arg? and typeof arg != 'undefined'
+      integer = parseInt(arg.toString())
+      return integer if isValidInteger(integer)
+    else
+      return null
 
 module.exports.ToQuantity = class ToQuantity extends Expression
   constructor: (json) ->
     super
 
   exec: (ctx) ->
     arg = @execArgs(ctx)
-    if arg? and typeof arg != 'undefined' then parseQuantity(arg.toString()) else null
+    if arg? and typeof arg != 'undefined'
+      quantity = parseQuantity(arg.toString())
+      return null if !quantity?
```

Comment:
Is this to ensure `null` is returned rather than `undefined`?

Reply:
No, I believe I was just trying to type up the CQL spec and didn't notice there could be a simplification.  I can remove this line, as parseQuantity won't ever return `undefined`